### PR TITLE
Feature/timeout processor

### PR DESCRIPTION
**Pull request by martingagnon**
## Description
Add [Timeout](http://reactivex.io/documentation/operators/timeout.html) processor

## Motivation and Context
This was needed to implement a Timeout feature

## How Has This Been Tested?
See attached tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: streams/src/nativeMain/kotlin/com/mirego/trikot/streams/reactive/processors/PreventFreezing.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import kotlin.native.concurrent.ThreadLocal
+import kotlin.native.concurrent.ensureNeverFrozen
+import kotlin.native.concurrent.isFrozen
+
+class OnMainTreadPublisher<T>(
```
Comment (marcantoinefortier) [inline streams/src/nativeMain/kotlin/com/mirego/trikot/streams/reactive/processors/PreventFreezing.kt]: Pourquoi cet ajout? Ça ne semble être utilisé nulle-part.

Comment (martingagnon) [reply streams/src/nativeMain/kotlin/com/mirego/trikot/streams/reactive/processors/PreventFreezing.kt]: Fixed!

File: streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublisherExtensions.kt [hunk 2]
```diff
@@ -110,6 +111,10 @@ fun <T, R> Publisher<T>.filterNotNull(block: ((T) -> R?)): Publisher<R> {
     return this.filter { block(it) != null }.map { block(it)!! }
 }
 
+fun <T> Publisher<T>.timeout(duration: Duration, message: String = "Default timeout message"): Publisher<T> {
```
Comment (npresseault) [inline streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublisherExtensions.kt]: Faudrait pouvoir passer un  timerFactory et defaulter à la FoundationConfiguration.timerFactory

Comment (martingagnon) [reply streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublisherExtensions.kt]: The processor has this configuration, its just the shorthand when you make your code (which I think will never specify the factory)

**Pull request merged**